static/app/components/codecov/container/container.tsx

```diff
@@ -0,0 +1,104 @@
+import {useState} from 'react';
+
+import {CodecovContext} from 'sentry/components/codecov/context/context';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+
+// Types
+export type CodecovContextTypes = {
+  branch: string | null;
+  codecovPeriod: string | null;
+  integratedOrg: string | null;
+  repository: string | null;
+};
+export type CodecovContextSetterTypes = {
+  setContextState: React.Dispatch<React.SetStateAction<CodecovContextTypes>>;
+  updateSelectorData: (value: Partial<CodecovContextTypes>) => void;
+};
+type CodecovUrlParams = 'repository' | 'integratedOrg' | 'branch' | 'codecovPeriod';
+
+type CodecovProviderProps = {
+  children?: React.ReactNode;
+};
```

Comment:
```suggestion
type CodecovProviderProps = {
  children: NonNullable<React.ReactNode>;
};
```

What would be the use case for using this provider without children?